### ISSUE TITLE
Many sites buggy when enabling option protect against detection based on Javascript

### ISSUE THREAD

**Issue by gelavat** [tarampampam/random-user-agent#600]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When I enable the option "Protect against detection based on Javascript", many sites are becoming buggy.
For example, when logging into github, the menu button does not show the menu at all when clicking on the icon. Other sites do not allow to login at all, and so forth.
I can turn this feature off and it works, but then the protection against Javascript based detection is not done, and hence the user-agent can still be retrieved by websites.

Are you sure that this feature is really only preventing the Javascript request about the user-agent and really this part only, and nothing else?
It might be that it is when using several extensions that this problem occurs, but if everything else is left untouched and is a pass-through for Javascript, except that only user-agent detection request, any other extension should still work very well.

### Steps to reproduce

1. Enable the "Protect against detection based on Javascript" option of the extension.
2. Login to github with your account
3. click on the menu to show it.

That should show the issue. If not, then maybe it is only when  some other extensions are also enabled.

### Browser

Firefox

### Extension version

4.1.1

### Anything else?

Please tell once you have been able to reproduce the issue. If not then we will have to figure out what extension, or version or whatever is producing this bug.
I use Firefox V128.0 and Windows 11.
Also this error happens since a few days only, and I did not touch this parameter in between so probably this is new, either due to a new Firefox version or other component.

**Comment by tarampampam** [comment]
Could you please provide what sites exactly not working properly, attach some screencasts maybe, or so on?

**Comment by gelavat** [comment]
Hello.
Sorry for the late answer.
Sure. The sites not working properly are many of them, almost 1 on 2, but only usually on sites where we have to login as a customer.
This happens already with Github.com.
As per your request, I made a video which shows on Github that the menu is not displayed at all when I click on it to display it. I clicked about 5 times on it in the video, and you will see that it is not displayed at all, nothing is shown:
https://github.com/user-attachments/assets/68ce1df8-7976-4aab-a73e-60a11c45eeb6

Yes it is possible that this is a conflict with another extension. I will be trying to isolate the other extension that would be causing this failure.
In any case, this happens only when I enable the "Protect against detection based on Javascript" option. This is 100% reproducible, and as soon as I disable this specific option, then all menus and sites are shown properly again, the issue disappears completely.

Thanks for any help.

**Comment by tarampampam** [comment]
Could you please temporarily disable all extensions except this one, and then enable them one by one to identify which one is causing the errors?

**Comment by gelavat** [comment]
Yes. I tested this, and I figured out that this happens especially when the CanvasBlocker from KKapsner (Fingerprint protection) is enabled while Random User-Agent is also enabled.
Also, turning on or off the privacy.resistFingerprinting Firefox parameter doesn't change anything.

As there are many options in this extension, I cannot tell if this is related to a specific option that I would have selected, which causes this issue to occur.
Can you please try to reproduce this on your side?

**Comment by JackeyLov5** [comment]
Hello I've been having this problem on one of my forums and it's causing me to make multiple purchases I haven't noticed it on any other sites yet but I think there's something wrong with it because I'm a mac and when I copy it always fails to copy it on some sites Here's a video that's been split into two because of the information involved The first one is where I'm commenting and he doesn't automatically refresh the page or the js is asynchronous but it doesn't happen when I disable the plugin.

https://github.com/user-attachments/assets/e808e6e4-2bf5-453f-ab42-4d2c136fdb77


https://github.com/user-attachments/assets/2830595d-c031-46f6-a376-ff8ad626e868

**Comment by gelavat** [comment]
Hello @tarampampam , have you been able to reproduce his issue with the conditions I provided ?
If not please tell and I will try to provide more configuration information to be able to reproduce it.

Thanks.
